src/main/java/rs/teslaris/core/controller/PatentController.java

```diff
@@ -0,0 +1,57 @@
+package rs.teslaris.core.controller;
+
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import rs.teslaris.core.annotation.Idempotent;
+import rs.teslaris.core.annotation.PublicationEditCheck;
+import rs.teslaris.core.dto.document.PatentDTO;
+import rs.teslaris.core.service.interfaces.document.PatentService;
+
+@RestController
+@RequestMapping("api/patent")
+@RequiredArgsConstructor
+public class PatentController {
+
+    private final PatentService patentService;
+
+    @GetMapping("/{publicationId}")
```

Comment:
use patentId instead of publicationId (this is comment for all cases in PatentController)

Reply:
Changed to documentId for better readability, it has to be the same in every occurence because of aspect reusability.